galaxy/api/serializers/serializers.py

```diff
@@ -346,6 +348,15 @@ def get_related(self, obj):
         if obj is None:
             return {}
         res = super(ImportTaskSerializer, self).get_related(obj)
+
+        if getattr(obj, 'collection'):
+            pk = obj.collection.namespace.pk
+            res.update({
+                'namespace': reverse('api:namespace_detail', kwargs={'pk': pk})
+            })
+        if getattr(obj, 'repository') is None:
+            return res
```

Comment:
if this block and the original block below now deal with two very different kinds of content, i think it warrants some comments detailing the different responsibilities. Its not clear just reading the code that lines 357-358 are collection specific, or that this implies the rest of the method is legacy repo specific.

Reply:
I will clean this up and make it more clear one block is for collections and the other repositories